src/detection/gpu/gpu_apple.m

```diff
@@ -56,8 +56,10 @@
             else if ([device supportsFeatureSet:MTLFeatureSet_macOS_GPUFamily1_v1])
                 ffStrbufSetStatic(&gpu->platformApi, "Metal Feature Set 1");
             #else // MAC_OS_X_VERSION_10_15
-            if ([device supportsFamily:MTLGPUFamilyMetal3])
-                ffStrbufSetStatic(&gpu->platformApi, "Metal 3");
+            if (@available(macOS 13.0, *)) {
```

Comment:
You are changing the logic. On macOS 13.0+, if the device doesn't support `MTLGPUFamilyMetal3` the code won't continue to check `MTLGPUFamilyCommon3`

Reply:
Sorry, you're right. I was trying to keep it a simple change. Looks like it didn't work with trying if logic shortcut `if (@available(macOS 13.0, *) && [device supportsFamily:MTLGPUFamilyMetal3])` so I just created a helper method for the common metal platforms and we check Metal 3 first on 13 and continue the fallback checks for others if it's not 13.

Reply:
I really don't like the change. Since it's just a warning, does `#pragma clang diagnostic ignored "-Wxxx"` work?

Reply:
My understanding of the availability checks are to prevent runtime crashes of this code being executed on a lower macos version. It is only a warning because the target SDK for compilation knows about the symbol.

Reply:
I reached out to someone with more domain expertise and suggested that https://developer.apple.com/documentation/metal/mtlgpufamily/mtlgpufamilymetal3 the symbol being a constant might not be a runtime crash, just an ugly compiler warning that could actually be fine... But, it looks like we use a `#ifndef MAC_OS_X_VERSION_10_15` wrapping calls that contain macos 13 features which isn't correct and we might be able to just bump that version check.

Reply:
> the symbol being a constant might not be a runtime crash

Definitely not. You can test it with `[device supportsFamily:(MTLGPUFamily)999999]`